src/EmotiBitPacket.cpp

```diff
@@ -363,7 +364,7 @@ int16_t EmotiBitPacket::getPacketElement(const String& packet, String& element,
 	return nextStartChar;
 }
 
-int16_t EmotiBitPacket::getPacketKeyedValue(const String &packet, const String &key, String &value, uint16_t startChar)
+int16_t EmotiBitPacket::getPacketKeyedValue(const String &packet, const String &key, String &value, int16_t startChar)
```

Comment:
For reviewer:

in case of end of packet, a `-1` returned by the called function (in this case `EmotiBitPacket::getPacketElement`) was being implicitly converted to 65535, which was created an infinite do-while loop in the code segment below.